amass/utils/misc.go

```diff
@@ -13,8 +13,9 @@ import (
 const (
 	// IPv4RE is a regular expression that will match an IPv4 address.
 	IPv4RE = "((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)[.]){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)"
+
 	// SUBRE is a regular expression that will match on all subdomains once the domain is appended.
-	SUBRE = "(([_a-zA-Z0-9]{1}[_a-zA-Z0-9-]{0,61}[a-zA-Z0-9]{1})[.]{1})+"
+	SUBRE = "(([a-zA-Z0-9]{1}|[_a-zA-Z0-9]{1}[_a-zA-Z0-9-]{0,61}[a-zA-Z0-9]{1})[.]{1})+"
```

Comment:
Regexes can be really hard to read and this makes them highly prone to errors. I've seen testing (and perhaps fuzzing) is the best way to avoid those hard to track bugs. It also helps to have a visual description of the regex, at work I've put a policy to always include a [Regex101](https://regex101.com/) prefilled playground.

Reply:
I definitely agree with your recommendation

Reply:
https://regex101.com/r/dSQH95/1

Reply:
awesome, it’s probably better to put it as a comment as well, so anyone reafing the code can immediately play with it. Add a test as well and it’s good to go 👍

Reply:
[SUBRE Regex101](https://regex101.com/r/dSQH95/1)